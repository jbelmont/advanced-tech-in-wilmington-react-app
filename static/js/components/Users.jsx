import React, { Component } from 'react';
import {browserHistory} from 'react-router';

import {addUserInfo} from '../actions/index';

const Users = ( { email, first_name, last_name, gender, id, props, onClick } ) => {

  const userDetail = event => {
    const element = Array.from(event.currentTarget.children);
    const userDetails = {
      email: element.filter(elem => elem["dataset"]["email"])[0].dataset["email"],
      firstName: element.filter(elem => elem["dataset"]["firstName"])[0].dataset["firstName"],
      lastName: element.filter(elem => elem["dataset"]["lastName"])[0].dataset["lastName"],
      gender: element.filter(elem => elem["dataset"]["gender"])[0].dataset["gender"],
      id: element.filter(elem => elem["dataset"]["id"])[0].dataset["id"]
    };
    onClick(userDetails);
    browserHistory.push({
      pathname: `/user/${userDetails["id"]}`,
      state: props
    });
  };

  return (
    <div className="users-container" onClick={event => userDetail(event)}>
        <span data-email={email} className="users-container-email">{email}</span>
        <span data-first-name={first_name} className="users-container-first-name">{first_name}</span>
        <span data-last-name={last_name} className="users-container-last-name">{last_name}</span>
        <span data-gender={gender} className="users-container-gender">{gender}</span>
        <span data-id={id} className="users-container-id">{id}</span>
    </div>
  )
}

export default Users;
