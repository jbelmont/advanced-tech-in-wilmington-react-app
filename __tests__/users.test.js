import React from 'react';
import Users from '../static/js/components/Users.jsx';
import renderer from 'react-test-renderer';

test('Test the <Users /> component', () => {
    const component = renderer.create(
        <Users  email="johnrambo@badass.net"
                first_name="John"
                last_name="Rambo"
                gender="Male"
                id="1"
                props=""
                onClick={() => {}}
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});