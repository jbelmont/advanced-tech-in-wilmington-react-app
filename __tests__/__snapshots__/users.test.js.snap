exports[`test Test the <Users /> component 1`] = `
<div
  className="users-container"
  onClick={[Function]}>
  <span
    className="users-container-email"
    data-email="johnrambo@badass.net">
    johnrambo@badass.net
  </span>
  <span
    className="users-container-first-name"
    data-first-name="John">
    John
  </span>
  <span
    className="users-container-last-name"
    data-last-name="Rambo">
    Rambo
  </span>
  <span
    className="users-container-gender"
    data-gender="Male">
    Male
  </span>
  <span
    className="users-container-id"
    data-id="1">
    1
  </span>
</div>
`;
